### ISSUE TITLE
S3218 should not report on generated files

### ISSUE THREAD

**Issue by ngm** [SonarSource/sonarlint-vs#608]
The [T4MVC](https://github.com/T4MVC/T4MVC) project generates a file, T4MVC.cs.  As per #164  classes within the file are annotated with `GeneratedCode` and `DebuggerNonUserCode` attributes, however with SonarLint 1.4.0 I am receiving *S3218 Rename this field to not shadow the outer class' member with the same name* analysis warnings.

``` C#
[GeneratedCode("T4MVC", "2.0"), DebuggerNonUserCode]
public static class Content {
    private const string URLPATH = "~/Content";
    public static string Url() { return T4MVCHelpers.ProcessVirtualPath(URLPATH); }
    public static string Url(string fileName) { return T4MVCHelpers.ProcessVirtualPath(URLPATH + "/" + fileName); }
    public static readonly string bootstrap_css = T4MVCHelpers.IsProduction() && T4Extensions.FileExists(URLPATH + "/bootstrap.min.css") ? Url("bootstrap.min.css") : Url("bootstrap.css");
             
    public static readonly string bootstrap_min_css = Url("bootstrap.min.css");
    [GeneratedCode("T4MVC", "2.0"), DebuggerNonUserCode]
    public static class css {
        private const string URLPATH = "~/Content/css";
        public static string Url() { return T4MVCHelpers.ProcessVirtualPath(URLPATH); }
        public static string Url(string fileName) { return T4MVCHelpers.ProcessVirtualPath(URLPATH + "/" + fileName); }
    }
    ...
```
It is the `URLPATH` field and the `Url()` methods of the inner `css` class that are highlighted with the warning.

**Comment by tvsonar** [comment]
Thank you for reporting this issue.
